wp-admin/js/widgets/media-widgets.js

```diff
@@ -73,6 +73,7 @@ wp.mediaWidgets = ( function( $ ) {
 		 * @type {Object}
 		 */
 		events: {
+			'click .notice a': 'selectMedia',
```

Comment:
I think the link should have a class name on it. If a subclass introduces a different error with a link in it, we'd want to make sure this event handler only handles clicks on the media library link. So either the container `. notice` element have a class like `missing-attachment-error`, or the link itself can have a `media-library` class that is then used in the selector here.